cmd/remove_integration_test.go

```diff
@@ -0,0 +1,538 @@
+package cmd
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/qri-io/qri/base/dsfs"
+	"github.com/spf13/cobra"
+)
+
+// RemoveTestRunner holds test info integration tests
+type RemoveTestRunner struct {
+	RepoRoot    *TestRepoRoot
+	Context     context.Context
+	ContextDone func()
+	TsFunc      func() time.Time
+	LocOrig     *time.Location
+	CmdR        *cobra.Command
+}
+
+// newRemoveTestRunner returns a new FSITestRunner.
+func newRemoveTestRunner(t *testing.T, peerName, testName string) *RemoveTestRunner {
+	root := NewTestRepoRoot(t, peerName, testName)
+
+	run := RemoveTestRunner{}
+	run.RepoRoot = &root
+	run.Context, run.ContextDone = context.WithCancel(context.Background())
+
+	// To keep hashes consistent, artificially specify the timestamp by overriding
+	// the dsfs.Timestamp func
+	counter := 0
+	run.TsFunc = dsfs.Timestamp
+	dsfs.Timestamp = func() time.Time {
+		counter++
+		return time.Date(2001, 01, 01, 01, 01, counter, 01, time.UTC)
+	}
+
+	// Set the location to New York so that timezone printing is consistent
+	location, err := time.LoadLocation("America/New_York")
+	if err != nil {
+		panic(err)
+	}
+	run.LocOrig = location
+	StringerLocation = location
+
+	return &run
+}
+
+// Delete cleans up after a RemoveTestRunner is done being used.
+func (run *RemoveTestRunner) Delete() {
+	dsfs.Timestamp = run.TsFunc
+	StringerLocation = run.LocOrig
+	run.ContextDone()
+	run.RepoRoot.Delete()
+}
+
+// ExecCommand executes the given command string
+func (run *RemoveTestRunner) ExecCommand(cmdText string) error {
+	run.CmdR = run.RepoRoot.CreateCommandRunner(run.Context)
+	return executeCommand(run.CmdR, cmdText)
+}
+
+// MustExec runs a command, returning standard output, failing the test if there's an error
+func (run *RemoveTestRunner) MustExec(cmdText string) string {
```

Comment:
not a huge fan of the way this `Must` pattern swallows the `t *testing.T` pointer as state. It makes this stuff really hard to use in benchmarks (which take a `*testing.B`) and subtests (which create new testing.T pointers). To me the test runner pattern shines when it's concise without being magical. I'd prefer:

```go
run, delete := newRemoveTestRunner(t, "test_peer", "example")
defer delete()

run.MustExec(t, "qri save --body=testdata/movies/body_two.json me/remove_test")
```